File: Java/GaussSeidel.java
// GaussSeidel.java
// Class for Gauss-Seidel Method in Java

public class GaussSeidel {

    public static double[] gaussSeidel(double[][] A, double[] b, double tolerance, int maxIterations) {
        int n = b.length;
        double[] x = new double[n]; // Solution vector initialized with zeros

        for (int k = 0; k < maxIterations; k++) {
            double[] xOld = x.clone(); // Copy of previous solution for convergence check

            for (int i = 0; i < n; i++) {
                double sum = b[i];
                
                for (int j = 0; j < n; j++) {
                    if (i != j) {
                        sum -= A[i][j] * x[j];
                    }
                }
                
                x[i] = sum / A[i][i];
            }

            // Check for convergence
            double error = 0.0;
            for (int i = 0; i < n; i++) {
                error += Math.abs(x[i] - xOld[i]);
            }

            if (error < tolerance) {
                return x; // Solution converged
            }
        }
        
        return null; // Solution did not converge within the given tolerance and iteration limit
    }
}

File: Python/gauss_seidel.py
# gauss_seidel.py
# Function for Gauss-Seidel Method in Python

import numpy as np

def gauss_seidel(A, b, tolerance=1e-6, max_iterations=100):
    """
    Solves the system of linear equations Ax = b using the Gauss-Seidel method.
    
    Parameters:
        A (ndarray): Coefficient matrix
        b (ndarray): Constants vector
        tolerance (float): Error tolerance for convergence
        max_iterations (int): Maximum number of iterations
    
    Returns:
        x (ndarray): Solution vector, or None if convergence is not achieved
    """
    n = len(b)
    x = np.zeros(n)  # Initial guess
    
    for k in range(max_iterations):
        x_old = x.copy()
        
        for i in range(n):
            sum_ = b[i]
            
            for j in range(n):
                if i != j:
                    sum_ -= A[i, j] * x[j]
            
            x[i] = sum_ / A[i, i]
        
        # Check for convergence
        error = np.linalg.norm(x - x_old, ord=np.inf)
        
        if error < tolerance:
            return x  # Solution converged
    
    return None  # Solution did not converge
